### DISCUSSION TITLE
TextEditor -- Centering text and eliminating the scrollbar.

### DISCUSSION THREAD

**Discussion by jtreagan** [fltk-rs/fltk-rs#1618] [Q&A]
I have a need for a one-line texteditor window that I can use to display and edit a title.  I'm sure a method exists that will center the text in the window, but I sure can't find it.  Could someone point me in the right direction?  Thanks.  Along with that problem is that the window pulls up a tiny scroll bar on the right edge.  I need to get rid of that.  I tried using  `set_scrollbar_size(0)`, but that didn't work.  Then my AI suggested I use a method called `hide_scrollbars()`, but that doesn't seem to exist in the fltk.rs universe.  So any advice as to how I can solve these problems?  Below, FYI, is the code I'm experimenting with:
```
use fltk::app::{set_font_size, App};
use fltk::enums::Color;
use fltk::prelude::{DisplayExt, GroupExt, WidgetExt};
use fltk::text::TextEditor;
use fltk::{text, window};
use fltk::window::Window;
use lib_myfltk::fltkutils::fltk_simple_editor;

fn main() {
    let app = App::default();

    let mut primarywin = Window::default().with_size(825, 900).with_pos(1000, 100);
    set_font_size(20);
    primarywin.set_color(Color::Blue);
    primarywin.make_resizable(true);

    let title = oneln_titled();

    primarywin.end();
    primarywin.show();

    app.run().unwrap();
}

pub fn oneln_titled() -> String {
    let mut buf = text::TextBuffer::default();
    set_font_size(40);
    buf.set_text("Untitled Question Bank");

    let mut simped = TextEditor::default().with_size(825, 40);
    simped.set_buffer(buf.clone());   // Clone is used here to avoid an ownership error.
    simped.wrap_mode(text::WrapMode::AtBounds, 0);
    simped.set_color(Color::White);
    simped.set_text_size(22);
    simped.set_text_color(Color::Black);
    simped.hide_scrollbars(); // Explicitly hide scrollbars
    buf.text()
}
```

**Comment by MoAlyousef** [comment]
Hi
You can use `simped.set_scrollbar_size(-1)` to hide the scrollbar.
That said, FLTK's Text Editor doesn't support rich text formatting, so it doesn't support centering text. It supports some [text attributes](https://docs.rs/fltk/latest/fltk/text/enum.TextAttr.html), but that's about it. You can use some trick like showing an input field then hiding it and displaying the inputted text in a Frame or Group with a centered alignment. I can post code if needed. The result isn't that great though. Otherwise, you would need to implement text formatting manually, which is not trivial.

**Comment by jtreagan** [comment]
@MoAlyousef 
> FLTK's Text Editor doesn't support rich text formatting

Did some searching and found this:

https://crates.io/crates/fltk-richtext/0.1.1

I haven't played around with it yet, which is next on my agenda, but have you worked with it any?

**Reply by jtreagan** [reply]
@MoAlyousef  Oh.  I see that you wrote this crate.  Cool!
